app/airq/models/clients.py

```diff
@@ -301,3 +305,22 @@ def maybe_notify(self) -> bool:
 
     def log_event(self, event_type: EventType, **event_data: typing.Any) -> Event:
         return Event.query.create(self.id, event_type, **event_data)
+
+    def get_last_client_event(self, exclude_type: EventType) -> Event:
```

Comment:
This should return `typing.Optional[Event]` since there might be no first event.